src/common/utils/resources.h

```diff
@@ -4,33 +4,204 @@
 #include <string>
 #include <atlstr.h>
 
-// Get a string from the resource file
-inline std::wstring get_resource_string(UINT resource_id, HINSTANCE instance, const wchar_t* fallback)
+#include <common/utils/language_helper.h>
+
+
+inline std::wstring get_english_fallback_string(UINT resource_id, HINSTANCE instance)
 {
-    wchar_t* text_ptr;
-    auto length = LoadStringW(instance, resource_id, reinterpret_cast<wchar_t*>(&text_ptr), 0);
-    if (length == 0)
+    // Try to load en-us string as the first fallback.
+    WORD english_language = MAKELANGID(LANG_ENGLISH, SUBLANG_ENGLISH_US);
+
+    ATL::CStringW english_string;
+    try
     {
-        // Try to load en-us string as the first fallback.
-        WORD english_language = MAKELANGID(LANG_ENGLISH, SUBLANG_ENGLISH_US);
-        ATL::CStringW english_string;
+        if (!english_string.LoadStringW(instance, resource_id, english_language))
+        {
+            return {};
+        }
+    }
+    catch (...)
+    {
+        return {};
+    }
+
+    return std::wstring(english_string);
+}
+
+inline std::wstring get_resource_string_language_override(UINT resource_id, HINSTANCE instance)
+{
+    std::wstring language = LanguageHelpers::load_language();
```

Comment:
If this is supposed to require a PowerToys restart, shouldn't we cache this result somehow? Thinking it's not good to do a file read and json parse on each time we want to get a resource.